certbot-cert/tasks/main.yml

```diff
@@ -33,14 +33,22 @@
 
 - name: Issue certbot certificate using DNS validation
   become: true
-  ansible.builtin.command: certbot certonly --test-cert --dns-cloudflare --dns-cloudflare-credentials /root/cloudflare.ini --noninteractive --agree-tos --email {{ certbot_email }} -d "{{ certbot_domains | join(',') }}" --dns-cloudflare-propagation-seconds {{ certbot_dns_validation_seconds }}
+  ansible.builtin.command: certbot certonly --dns-cloudflare --dns-cloudflare-credentials /root/cloudflare.ini --noninteractive --agree-tos --email {{ certbot_email }} -d "{{ certbot_domains | join(',') }}" --dns-cloudflare-propagation-seconds {{ certbot_dns_validation_seconds }}
   args:
     creates: /etc/letsencrypt/live/{{ certbot_domains | first }}/privkey.pem
   when: certbot_validation_strategy == "cloudflare"
 
+- name: Ensure webroot directory exists
```

Comment:
Don't we need this to happen in the Nginx role since Nginx is including the snippet and default that uses this webroot?

Reply:
they  both need it. Certbot has to be able to write here as well. Maybe we should add to both places, to be safest

Reply:
Yeah, I think having it both places is the safest solution.  It is possible we'd be using Nginx without certbot, and I guess it is also possible we'd use certbot without Nginx (even though this is much less likely and we have the certbot role have an nginx dependency), so seems to me they both need to stand on their own.